pkg/ingester/ingester.go

```diff
@@ -1696,27 +1696,42 @@ func (i *Ingester) labelNamesCommon(ctx context.Context, req *client.LabelNamesR
 
 // MetricsForLabelMatchers returns all the metrics which match a set of matchers.
 func (i *Ingester) MetricsForLabelMatchers(ctx context.Context, req *client.MetricsForLabelMatchersRequest) (*client.MetricsForLabelMatchersResponse, error) {
-	result, cleanup, err := i.metricsForLabelMatchersCommon(ctx, req)
+	result := &client.MetricsForLabelMatchersResponse{}
+	cleanup, err := i.metricsForLabelMatchersCommon(ctx, req, func(l labels.Labels) error {
+		result.Metric = append(result.Metric, &cortexpb.Metric{
+			Labels: cortexpb.FromLabelsToLabelAdapters(l),
+		})
+		return nil
+	})
 	defer cleanup()
 	return result, err
 }
 
 func (i *Ingester) MetricsForLabelMatchersStream(req *client.MetricsForLabelMatchersRequest, stream client.Ingester_MetricsForLabelMatchersStreamServer) error {
-	result, cleanup, err := i.metricsForLabelMatchersCommon(stream.Context(), req)
+	result := &client.MetricsForLabelMatchersStreamResponse{}
+
+	cleanup, err := i.metricsForLabelMatchersCommon(stream.Context(), req, func(l labels.Labels) error {
+		result.Metric = append(result.Metric, &cortexpb.Metric{
+			Labels: cortexpb.FromLabelsToLabelAdapters(l),
+		})
+
+		if len(result.Metric) > metadataStreamBatchSize {
```

Comment:
```suggestion
		if len(result.Metric) >= metadataStreamBatchSize {
```
more than equal may be right..?

Reply:
thxs